### ISSUE TITLE
iSCSI disk fails to attach to application pod 

### ISSUE THREAD

**Issue by ksatchit** [openebs/openebs#1076]
## Is this a BUG REPORT or FEATURE REQUEST?
BUG REPORT

**What happened**:

A percona deployment as part of the CI failed with the following error : 

```
  FirstSeen    LastSeen    Count    From            SubObjectPath    Type        Reason            Message
  ---------    --------    -----    ----            -------------    --------    ------            -------
  23m        23m        1    default-scheduler            Normal        Scheduled        Successfully assigned percona to hch-01
  23m        23m        1    kubelet, hch-01                Normal        SuccessfulMountVolume    MountVolume.SetUp succeeded for volume "default-token-2317t" 
  21m        21m        1    kubelet, hch-01                Warning        FailedMount        MountVolume.SetUp failed for volume "pvc-bd2faec6-eae5-11e7-9f0a-02b983f0a4db" : failed to get any path for iscsi disk, last err seen:
Could not attach disk: Timeout after 10s
  21m    1m    10    kubelet, hch-01        Warning    FailedMount    Unable to mount volumes for pod "percona_percona(bd23c77e-eae5-11e7-9f0a-02b983f0a4db)": timeout expired waiting for volumes to attach/mount for pod "percona"/"percona". list of unattached/unmounted volumes=[demo-vol1]
  21m    1m    10    kubelet, hch-01        Warning    FailedSync    Error syncing pod
  21m    1m    17    kubelet, hch-01        Warning    FailedMount    MountVolume.SetUp failed for volume "pvc-bd2faec6-eae5-11e7-9f0a-02b983f0a4db" : failed to get any path for iscsi disk, last err seen:
iscsi: failed to attach disk: Error: iscsiadm: default: 1 session requested, but 1 already present.
iscsiadm: Could not log into all portals
```
There are following messages from /var/log/syslog: 

```
Dec 27 15:25:47 MayaHost01 kernel: [109730.382253] sd 4:0:0:0: lun278725195042828 has a LUN larger than allowed by the host adapter
```
The device is listed in fdisk -l , but the mount is seen to fail

**What you expected to happen**:

iSCSI disk should be attached to the application pod successfully 

**How to reproduce it (as minimally and precisely as possible)**:

Deploy any application with openebs storageclass using "latest" tagged maya/provisioner/jiva

**Anything else we need to know?**:

**Comment by ksatchit** [comment]
Here is some initial analysis from @payes : 

"It is a problem with LUN number, initially LUN 1 was being used which has been changed to LUN 0 in gotgt"

**Comment by ksatchit** [comment]
O/P of iscsiadm -m session -P 3 w/ jiva image 0.5.0 & jiva latest (note the difference in the LUN numbers) : 

0.5.0
-----
: 
************************
Attached SCSI devices:
************************
Host Number: 3  State: running
scsi3 Channel 00 Id 0 Lun: 1
              Attached scsi disk sdb          State: running

latest: 
------
************************
 Attached SCSI devices:
************************
Host Number: 11 State: running
scsi11 Channel 00 Id 0 Lun: 0
             Attached scsi disk sdb          State: running

**Comment by ksatchit** [comment]
This is fixed in openebs-provisioner image to 0.5.1-RC2
